jobs/jobs/utils.py

```diff
@@ -266,6 +269,20 @@ def files_data_to_pipeline_arg(
         )
 
 
+async def fill_s3_signed_url(files: List[pipeline.PipelineFile]):
+    async def fill(file):
+        file.s3_signed_url = await create_pre_signed_s3_url(
+            bucket=file.bucket, path=file.input_path
+        )
+
+    if not JOBS_RUN_PIPELINES_WITH_SIGNED_URL:
+        return files
+
+    tasks = [fill(f) for f in files]
+    await asyncio.gather(*tasks)
```

Comment:
I think we don't need it to be run in parallel here, it's ok to run 1 by 1, but `await` results to allow other `coroutines` do their job

Reply:
could you please explain why you think so?